zcutil/make-release.py

```diff
@@ -330,6 +349,37 @@ def sh_log(*args):
         raise SystemExit('Nonzero exit status: {!r}'.format(status))
 
 
+def sh_progress(markers, *args):
+    try:
+        import progressbar
+    except:
+        sh_log(*args)
+        return
+
+    PIPE = subprocess.PIPE
+    try:
+        p = subprocess.Popen(args, stdout=PIPE, stderr=PIPE, stdin=None)
+    except OSError:
+        logging.error('Error launching %r...', args)
+        raise
+
+    pbar = progressbar.ProgressBar(max_value=len(markers))
+    marker = 0
+    pbar.update(marker)
+    logging.debug('Run (log PID %r): %r', p.pid, args)
+    for line in p.stdout:
+        logging.debug('> %s', line.rstrip())
+        for idx, val in enumerate(markers[marker:]):
+            if val in line:
+                marker += idx + 1
```

Comment:
Should this be `marker = idx + 1`? It seems weird to accumulate (`+=`) with an index.

Reply:
Nevermind. I see that `idx` is 0-indexes on the slice, starting at `marker`, so `marker + idx + 1` will be the absolute index just after the `val` marker.

Reply:
Yep. I wanted to ensure that if the build order changed, the progress bar did not get confused (ie. it always makes progress forwards, not backwards).